examples/step-50/doc/intro.dox

```diff
@@ -1,90 +1,128 @@
-for optimal multigrid complexity, it leads to problems when defining
-where to smooth and what boundary conditions to pose for the operators
-defined on individual levels if the level boundary is not an external
-boundary. These questions are discussed in detail in the
-@ref mg_paper "Multigrid paper by Janssen and Kanschat" that describes
-the implementation in deal.II.
 
+<h3>Workload imbalance</h3>
+For the active mesh, we use the parallel::distributed::Triangulation class as done
+in step-40 which uses functionality in the external library
+<a href="http://www.p4est.org/">p4est</a> for the distribution of the active cells
+among processors. For the non-active cells in the multilevel hierarchy, deal.II
+implements what we will refer to as the ``first-child rule'' where, for each cell
+in the hierarchy, we recursively assign the parent of a cell to the owner of the
+first child cell. The following figures give an example of such a distribution. Here
+the left image represents the active cells for a sample 2D mesh partitioned using a
+space-filling curve (similar to p4est), the center image gives the tree representation
+of the active mesh, and the right image gives the multilevel hierarchy of cells. The
+colors and numbers represent the different processors. The circular nodes in the tree
+are the non-active cells which are distributed using the ``first-child rule''.
 
+<img src="https://www.dealii.org/images/steps/developer/step-50-workload-example.png" alt="">
 
-<h3>The testcase</h3>
+Included among the output to screen in this example is a value ``Partition efficiency''
+given by 1/MGTools::workload_imbalance(). This value, which will be denoted
+by $\mathbb{E}$,  quantifies the overhead produced by not having a perfect work balance
+on each level of the multigrid hierarchy (as is evident from the example above).
 
-The problem we solve here is exactly the same as in
-step-6, the only difference being the solver we use
-here. You may want to look there for a definition of what we solve,
-right hand side and boundary conditions. Obviously, the program would
-also work if we changed the geometry and other pieces of data that
-defines this particular problem.
+For defining $\mathbb{E}$, it is important to note that, as we are using local smoothing
+to define the multigrid hierarchy (see the @ref mg_paper "multigrid paper" for a description of
+local smoothing), the refinement level of a cell corresponds to that cells multigrid
+level. Now, let $N_{\ell}$ be the number of cells on level $\ell$
+(both active and non-active cells) and $N_{\ell,p}$ be the subset owned by process
+$p$. Assuming that the workload for any one processor is proportional to the number
+of cells owned by that processor, the optimal workload per processor is given by
+@f{align*}
+W_{\text{opt}} = \sum_{\ell}\frac1{n_{p}}\sum_{p}N_{\ell,p}=\frac1{n_{p}}\sum_{\ell} N_{\ell}.
+@f}
+Next, assuming a synchronization of work on each level (i.e., on each level of a vcycle,
```

Comment:
I think we typically write *V-cycle* - do you agree?
```suggestion
Next, assuming a synchronization of work on each level (i.e., on each level of a V-cycle,
```
You might best use find-and-replace so I do not annotate all those uses below.

Reply:
Yes, I agree. Actually not sure why I got away from this..